### PR TITLE
Add filter: ignore if unmanaged by release-manager

### PR DESCRIPTION
**Pull request by JacobValdemar**
I moved the interaction with the release-manager into a function `retrieveFromReleaseManager`, which can be called when I need to make a request to the release-manager.

I also moved the types related to responses from the release-manager into a file `types.go`. I wanted to refer the types directly via `import httpinternal "github.com/lunarway/release-manager/internal/http"`, but apparently this dependency requires a complex dependency to k8s, which created chaos in `go.mod`. Thus this copy-paste solution.

Manually tested against a fake webhook payload from an unmanaged repository (without it being marked as an ignored repository).

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -10,7 +10,8 @@ require (
 	github.com/pkg/errors v0.9.1
 	github.com/rs/zerolog v1.19.0
 	github.com/spf13/pflag v1.0.5
-	github.com/stretchr/testify v1.4.0
+	github.com/stretchr/testify v1.5.1
 	goji.io v2.0.2+incompatible
 	gopkg.in/yaml.v2 v2.2.8
+
```
Comment (Crevil) [inline go.mod:16]: Newline should not be added :) This is usually a machine controlled file :) 
```suggestion
```

File: types.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
+package main
+
+import "time"
+
+// Would be nice to be able to do, instead of this:
+// httpinternal "github.com/lunarway/release-manager/internal/http"
```
Comment (Crevil) [inline types.go:6]: The `internal` package is special in Go and specifically prohibits imports from other modules.
We could however move them out into a shared package in `release-manager` to make it possible. :)

Comment (JacobValdemar) [reply types.go:6]: Oh so that's why it wouldn't work

**Comment by JacobValdemar** [comment]
> Looks good. Good with the extraction of the release manager HTTP requests.
> 
> A more contained implementation would maybe be a new `ReleaseManagerClient` with methods `Policies` and `Describe`. That way you lock the relation ship between the endpoint and the response types as this is no hidden inside the methods.
> 
> Further you can configure the client with the auth token and base bath, so that it gets out of the way of the handler implementation :)
> 
> Maybe in another PR just to keep it simple :)

Good idea, but in that case wouldn't it make more sense to create this ReleaseManagerClient inside the release-manager (server) repo? That way other users of the server can make use of it.

**Comment by Crevil** [comment]
It is unusual in Go as the HTTP API is already an API for clients. :) Some services do expose it as an SDK with a meaningful abstraction around the endpoints (Amazon AWS SDK).
Another option is to provide the API schema instead and then generate a client from that (OpenAPI). That spproach makes it possible to generate the client in any language instead.

As long as it is as basic as this, I would be reluctant to provide and support the client along with the server.

**Pull request merged**